### PR TITLE
#3128563 - Add Import service for calling the 'get' endpoint.

### PR DESCRIPTION
**Pull request by shabananavas**

### CODE REVIEW

**Comment by krystalcode** [comment]
@shabananavas Assigning this back to you as this is still missing the actual implementation in the import manager.

File: src/Client/ClientInterface.php [hunk 1]
```diff
@@ -39,4 +39,15 @@ interface ClientInterface {
...
+   * Gets the resource's main entity by its primary ID.
+   *
+   * @param int|string $id
+   *   The ID of the entity to get.
+   *
+   * @return object
+   *   The parsed response.
+   */
+  public function get($id): object;
+
```
Comment (krystalcode) [inline src/Client/ClientInterface.php:54]: **CHANGE REQUEST**

We haven't decided to use PHP 7.2+ syntax for method responses in this module yet, so let's remove that. Also, the expected response here would be the entity object i.e. " The parsed response" => "The remote entity object".

File: src/Import/ManagerInterface.php [hunk 2]
```diff
@@ -20,6 +20,22 @@ interface ManagerInterface {
...
+   *
+   * @param string $sync_type_id
+   *   The ID of the entity sync type.
+   * @param \Drupal\Core\Entity\EntityInterface $drupal_entity
+   *   The Drupal entity that we are trying to sync.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface
+   *   The Drupal entity that was synced.
+   */
+  public function syncGet($sync_type_id, EntityInterface $drupal_entity);
```
Comment (krystalcode) [inline src/Import/ManagerInterface.php]: **CHANGE REQUEST**

As discussed, this should be `public function importLocalEntity`.

Comment (shabananavas) [reply src/Import/ManagerInterface.php]: Yeah, I didn't get to working on this yesterday.

**Comment by shabananavas** [comment]
Okay, I think this is now ready for review.

**Comment by krystalcode** [comment]
Merged on Drupal.org

**Pull request closed without merging**